source/common/common/logger.cc

```diff
@@ -55,29 +58,50 @@ void DelegatingLogSink::set_formatter(std::unique_ptr<spdlog::formatter> formatt
 
 void DelegatingLogSink::log(const spdlog::details::log_msg& msg) {
   absl::ReleasableMutexLock lock(&format_mutex_);
-  if (!formatter_) {
-    lock.Release();
-    sink_->log(absl::string_view(msg.payload.data(), msg.payload.size()));
-    return;
-  }
+  absl::string_view msg_view = absl::string_view(msg.payload.data(), msg.payload.size());
 
+  // This memory buffer must exist in the scope of the entire function,
+  // otherwise the string_view will refer to memory that is already free.
   fmt::memory_buffer formatted;
-  formatter_->format(msg, formatted);
+  if (formatter_) {
+    formatter_->format(msg, formatted);
+    msg_view = absl::string_view(formatted.data(), formatted.size());
+  }
   lock.Release();
-  sink_->log(absl::string_view(formatted.data(), formatted.size()));
+
+  if (should_escape_) {
+    // Split the actual log message from the trailing whitespace.
+    auto eol_it = std::find_if_not(msg_view.rbegin(), msg_view.rend(), absl::ascii_isspace);
```

Comment:
What happens if there is no whitespace at the end of the string? You may be doing more with rend() than you are supposed to be below?

Reply:
I added unit tests to verify nothing is appended when there are no whitespaces at the end. I believe the code will just get a substring of size 0 for the `msg_whitespace_view`

Reply:
Added fuzz tests!